iml-gui/crate/src/components/command_modal.rs

```diff
@@ -284,3 +547,643 @@ fn close_button() -> Node<Msg> {
+                        ls[1] = is_subset(js, cmd.deps());
+                    }
+                    (Some(_), _) => {
+                        ls[0] = true;
+                    }
+                    (_, _) => {}
+                }
+            }
+            Select::CommandJobSteps(i, js, ks) => {
+                let cmd0 = self.commands.get(i);
+                let jobs0 = if is_subset(js, &job_ids) {
+                    Some(js.iter().map(|j| Arc::clone(&self.jobs[j])).collect::<Vec<_>>())
+                } else {
+                    None
+                };
+                let steps0 = if is_subset(ks, &step_ids) { Some(()) } else { None };
+                match (cmd0, jobs0, steps0) {
+                    (Some(cmd), Some(jobs), Some(_)) => {
+                        ls[0] = true;
+                        ls[1] = is_subset(js, cmd.deps());
+                        let ids = jobs.iter().flat_map(|x| x.deps()).copied().collect::<Vec<u32>>();
+                        ls[2] = is_subset(ks, &ids);
+                    }
+                    (Some(cmd), Some(_), _) => {
+                        ls[0] = true;
+                        ls[1] = is_subset(js, cmd.deps());
+                    }
+                    (Some(_), _, _) => {
+                        ls[0] = true;
+                    }
+                    (_, _, _) => {}
+                }
+            }
+        }
+        // make ensure the consistency levels are ordered
+        if ls[0] && ls[1] && ls[2] {
+            (true, true, true)
+        } else if ls[0] && ls[1] {
+            (true, true, false)
+        } else if ls[0] {
+            (true, false, false)
+        } else {
+            (false, false, false)
+        }
+    }
+
+    fn refresh_view(&mut self, layers: (bool, bool, bool)) {
+        let (cmds_ok, jobs_ok, steps_ok) = layers;
+        if cmds_ok {
+            self.commands_view = convert_to_sorted_vec(&self.commands);
+        }
+        if jobs_ok {
+            let ids = extract_sorted_keys(&self.jobs);
+            let jobs_graph_data = ids
+                .iter()
+                .map(|k| RichJob::new(self.jobs[k].inner.clone(), extract_wait_fors_from_job))
+                .collect::<Vec<RichJob>>();
+            self.jobs_graph = build_direct_dag(&jobs_graph_data);
+        }
+        if steps_ok {
+            self.steps_view = convert_to_sorted_vec(&self.steps);
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rand_core::{RngCore, SeedableRng};
+    use rand_xoshiro::Xoroshiro64Star;
+
+    #[derive(Default, Clone, Debug)]
+    struct Db {
+        all_cmds: Vec<Command>,
+        all_jobs: Vec<Job0>,
+        all_steps: Vec<Step>,
+    }
+
+    impl Db {
+        fn select_cmds(&self, is: &[u32]) -> Vec<Command> {
+            self.all_cmds
+                .iter()
+                .filter(|x| is.contains(&x.id))
+                .map(|x| x.clone())
+                .collect()
```

Comment:
```suggestion
        fn select_cmds(&self, is: &[u32]) -> Vec<&Command> {
            self.all_cmds
                .iter()
                .filter(|x| is.contains(&x.id))
                .collect()
```

We should not need to clone here.

Reply:
Let's continue to discuss it there: https://github.com/whamcloud/integrated-manager-for-lustre/pull/1796/files#r417082999